executioner/languages/docker.ts

```diff
@@ -0,0 +1,170 @@
+  filename: string;
+  content: string;
+  metadata?: any;
+};
+
+const IMAGE_BASE = 'ghcr.io/42dotmk/colosseum-executioner-';
+const WORKDIR = process.env.WORKDIR || '_work';
+
+if (!fs.existsSync(WORKDIR)) {
+  fs.mkdirSync(WORKDIR);
+}
+
+function parseDuration(duration: string) {
+  const match = duration.match(/(\d+)m(\d+(?:\.\d+)?)s/);
+  if (!match) return null;
+
+  const minutes = Number(match[1]);
+  const seconds = Number(match[2]);
+
+  return minutes * 60 + seconds;
+}
+
+export const execute = async (files: File[], input: File[], options: LanguageOptions) => {
+  const id = v4();
+
+  const subWorkspace = path.join(WORKDIR, id);
+  const srcDir = path.resolve(path.join(subWorkspace, "src"));
+  const inputDir = path.resolve(path.join(subWorkspace, "input"));
+  const outputDir = path.resolve(path.join(subWorkspace, "output"));
+  const lang = options.language;
+
+  if (!fs.existsSync(subWorkspace)) {
+    fs.mkdirSync(subWorkspace);
+  }
+
+  if (!fs.existsSync(srcDir)) {
+    fs.mkdirSync(srcDir);
+  }
+
+  if (!fs.existsSync(inputDir)) {
+    fs.mkdirSync(inputDir);
+  }
+
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(outputDir);
+  }
+
+  const timeFilename = `time`;
+  const stdoutFilename = `stdout`;
+  const compileStdoutFilename = `compile.stdout`;
+  const compileStderrFilename = `compile.stderr`;
+  const stderrFilename = `stderr`;
+
+  const timePath = path.resolve(path.join(subWorkspace, timeFilename));
+  const stdoutPath = path.resolve(path.join(subWorkspace, stdoutFilename));
+  const stderrPath = path.resolve(path.join(subWorkspace, stderrFilename));
+  const compileStdoutPath = path.resolve(path.join(subWorkspace, compileStdoutFilename));
+  const compileStderrPath = path.resolve(path.join(subWorkspace, compileStderrFilename));
+
+  for (const file of files) {
+    const filePath = path.resolve(path.join(srcDir, file.filename));
+    await writeFile(filePath, file.content);
+  }
+
+  for (const file of input) {
+    const filePath = path.resolve(path.join(inputDir, file.filename));
+    await writeFile(filePath, file.content);
+  }
+
+  try {
+    return await new Promise(async (resolve) => {
+      await writeFile(timePath, "");
+      await writeFile(stdoutPath, "");
+      await writeFile(stderrPath, "");
+      await writeFile(compileStdoutPath, "");
+      await writeFile(compileStderrPath, "");
+
+      const args = [
+        'run',
+        "--rm",
+        "-v",
+        `${srcDir}:/exc/src`,
+        "-v",
+        `${inputDir}:/exc/input`,
+        "-v",
+        `${outputDir}:/exc/output`,
+        "-v",
+        `${timePath}:/exc/${timeFilename}`,
+        "-v",
+        `${stdoutPath}:/exc/${stdoutFilename}`,
+        "-v",
+        `${stderrPath}:/exc/${stderrFilename}`,
+        "-v",
+        `${compileStdoutPath}:/exc/${compileStdoutFilename}`,
+        "-v",
+        `${compileStderrPath}:/exc/${compileStderrFilename}`,
+        "-i",
+        `${IMAGE_BASE}${lang}`
+      ];
```

Comment:
Please add resource limits, to prevent infinite loops and memory leaks https://docs.docker.com/config/containers/resource_constraints/